react/features/base/tracks/subscriber.js

```diff
@@ -0,0 +1,31 @@
+// @flow
+
+import _ from 'lodash';
+
+import { StateListenerRegistry } from '../../base/redux';
+
+declare var APP: Object;
+
+/**
+ * Stores the list of currently sharing particpant ID's.
+ */
+let currentlySharingParticipants = [];
+
+/**
+ * Notifies when the list of currently sharing participants changes.
+ */
+StateListenerRegistry.register(
+    /* selector */ state => state['features/base/tracks'],
+    /* listener */ tracks => {
+        if (typeof APP !== 'object') {
+            return;
+        }
+
+        const sharingParticipants = tracks.filter(tr => tr.videoType === 'desktop').map(t => t.participantId);
+
+        if (!_.isEqual(_.sortBy(currentlySharingParticipants), _.sortBy(sharingParticipants))) {
+            currentlySharingParticipants = sharingParticipants;
```

Comment:
Please don't store tracks there, the listener function also returns the previous selector value, you can use that instead, right?

Reply:
well yes but then we need query for CS tracks on both previous and current tracks, which sound a bit more costly?

Reply:
Only if they change, so not that costly I think.

Reply:
refactored as discussed